src/pymor/reductors/neural_network.py

```diff
@@ -345,17 +397,31 @@ def _compute_sample(self, mu, u=None):
 
         product = self.pod_params.get('product')
 
-        return [(mu, self.reduced_basis.inner(u, product=product)[:, 0])]
+        # conditional expression to check for instationary solution to return self.nt solutions
+        assert hasattr(self, 'nt')
+        parameters = [mu.with_(t=t) for t in np.linspace(0, self.T, self.nt)] if not self.is_stationary else [mu]
+        samples = [(mu, self.reduced_basis.inner(u_t, product=product)[:, 0]) for mu, u_t in
```

Comment:
Here we actually assume that `self.reduced_basis` is an orthonormal basis of the reduced space. Since it is allowed to pass the reduced basis as an argument to the reductor now, this might not be ensured. We should either check this or rewrite this line accordingly.

To be honest, this is another line of code that feels wrong to me (please don't misunderstand me, it was the same before and I refer to the general concept how these reductors work). From my point of view, it would be nice if the reductor knew nothing about the high-dimensional states and thus also does not project them onto the reduced basis. The reduced basis would only be required for reconstruction. How do you feel about this @TiPlath? But definitely nothing to address in this PR. Maybe we should sit together (also with @sdrave and @ftschindler) and try to redesign the reductors (also having the adaptive model hierarchies in mind).

Reply:
I like the idea, happy to discuss.
To get this right, you are saying that we should split reduced basis construction and neural network approximations, such that the neural network only knows about the reduced order model? 
Thinking further, then it would also get rid of the ```fom is None``` case or am I wrong?

Reply:
Some arguments for having this inside the reductor: 
- it is called "reductor" because it should reduce the space, should we think about calling it different then?
- I like right now that I do not have to know anything about pyMOR to use these reductors because it will do all the data handling for me. I just have to pass in my simulation data. This is very (new) user friendly to me.
- ... to be continued

arguments against:
- complex class structure which does not serve a single purpose (which is not nice to have for object oriented coding)
- ... to be continued